youtube/feed.go

```diff
@@ -276,39 +277,34 @@ var (
 	ErrMaxCustomMessageLength = errors.New("max length of custom message can be 500 chars")
 )
 
-func (p *Plugin) parseYtUrl(url string) (t ytUrlType, id string, err error) {
-	if ytUrlShortRegex.MatchString(url) {
-		capturingGroups := ytUrlShortRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][2]) > 0 {
-			return ytUrlTypeVideo, capturingGroups[0][2], nil
-		}
-	} else if ytVideoUrlRegex.MatchString(url) {
-		capturingGroups := ytVideoUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][4]) > 0 {
-			return ytUrlTypeVideo, capturingGroups[0][4], nil
-		}
-	} else if ytChannelUrlRegex.MatchString(url) {
-		capturingGroups := ytChannelUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeChannel, capturingGroups[0][5], nil
-		}
-	} else if ytCustomUrlRegex.MatchString(url) {
-		capturingGroups := ytCustomUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeCustom, capturingGroups[0][5], nil
-		}
-	} else if ytUserUrlRegex.MatchString(url) {
-		capturingGroups := ytUserUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeUser, capturingGroups[0][5], nil
-		}
-	} else if ytHandleUrlRegex.MatchString(url) {
-		capturingGroups := ytHandleUrlRegex.FindAllStringSubmatch(url, -1)
-		if len(capturingGroups) > 0 && len(capturingGroups[0]) > 0 && len(capturingGroups[0][5]) > 0 {
-			return ytUrlTypeHandle, capturingGroups[0][5], nil
-		}
+func (p *Plugin) parseYtUrl(url string) (idType ytUrlType, id string, err error) {
+	if id = p.getIDFromRegexUrl(ytUrlShortRegex, url, 2); id != "" {
+		idType = ytUrlTypeVideo
+	} else if id = p.getIDFromRegexUrl(ytVideoUrlRegex, url, 4); id != "" {
+		idType = ytUrlTypeVideo
+	} else if id = p.getIDFromRegexUrl(ytChannelUrlRegex, url, 5); id != "" {
+		idType = ytUrlTypeChannel
+	} else if id = p.getIDFromRegexUrl(ytCustomUrlRegex, url, 5); id != "" {
+		idType = ytUrlTypeCustom
+	} else if id = p.getIDFromRegexUrl(ytUserUrlRegex, url, 5); id != "" {
+		idType = ytUrlTypeUser
+	} else if id = p.getIDFromRegexUrl(ytHandleUrlRegex, url, 5); id != "" {
+		idType = ytUrlTypeHandle
+	} else {
+		idType = ytUrlTypeInvalid
+		err = errors.New("invalid or incomplete url")
+	}
+
+	return idType, id, err
```

Comment:
I am unconvinced the naked return pulls its weight here; what does this buy us over the simpler
```go
if id := p.getIDFromRegexUrl(...); id != "" {
	return ytUrlTypeVideo, id, nil
} else if ... {

}
...
return ytUrlTypeInvalid, "", errors.New("invalid or incomplete url")
```
?

Reply:
It minimises the number of elements which are repeated, and as named return values are being used, just a plain `return` statement can be used with this structure, which would implicitly return the values set for the named variables.

Reply:
I'm aware of what naked returns are. My point is that just because they *can* be used does not mean that they *should* be used. See the [relevant section in the Go wiki](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters), which advises

> Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values.

Reply:
Ok, I can revert switching to a naked return,

Reply:
Do you intend to do that anytime soon @Wolveric ?